lib/spack/spack/package_base.py

```diff
@@ -433,7 +449,7 @@ def remove_files_from_view(self, view, merge_map):
 Pb = TypeVar("Pb", bound="PackageBase")
 
 
-class PackageBase(WindowsRPath, PackageViewMixin, metaclass=PackageMeta):
+class PackageBase(WindowsRPath, PackageViewMixin, BinaryPackage, metaclass=PackageMeta):
```

Comment:
Could you make this an optional class that a package could inherit from?

Reply:
I think this should go together with `%gcc`, so anything that has a compiler need to have that. We'll probably relax the set of nodes that need to have a compiler later (e.g. no need to have it for pure Python code, etc.).

Reply:
I think it would work better to explicitly add the dependence on gcc-runtime.

Reply:
Either directly with depends_on('gcc-runtime') or by setting a property of the Package class.

Reply:
Just concretize with --fresh once and be done with it?

Reply:
That does not seem to override the environment setting from what I can tell with `spec -c edges` in the environment.